File: 백준/Bronze/15593. Lifeguards （Bronze）/Lifeguards （Bronze）.py
n = int(input()) #1 ~ 100

time = [0] * 1001
arr = []
for _ in range(n):
    tmp = list(map(int, input().split()))
    arr.append(tmp)
    for i in range(tmp[0], tmp[1]):
        time[i] += 1

ans = 0
for tmp in arr:
    for i in range(tmp[0], tmp[1]):
        time[i] -= 1 # 빼주기
    ans = max(ans, 1001 - time.count(0))
    for i in range(tmp[0], tmp[1]):
        time[i] += 1 #다음 계산을 위해 다시 되돌려놓기

print(ans)

File: 백준/Bronze/15593. Lifeguards （Bronze）/Lifeguards （Bronze）.java
import java.io.*;
import java.util.*;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st;

        int n = Integer.parseInt(br.readLine());
        int[][] arr = new int[n][2];
        int[] time = new int[1001];
        for (int i = 0; i < n; i++) {
            st = new StringTokenizer(br.readLine());
            arr[i][0] = Integer.parseInt(st.nextToken());
            arr[i][1] = Integer.parseInt(st.nextToken());
            for (int j = arr[i][0]; j < arr[i][1]; j++) {
                time[j] += 1;
            }
        }
        int ans = 0;
        for (int i = 0; i < n; i++) {
            for (int j = arr[i][0]; j < arr[i][1]; j++) {
                time[j] -= 1;
            }
            int cnt = 0;
            for (int j = 0; j < 1001; j++) {
                if(time[j] > 0) cnt++;
            }

            for (int j = arr[i][0]; j < arr[i][1]; j++) {
                time[j] += 1;
            }
            ans = Math.max(ans, cnt);
        }

        System.out.println(ans);
    }
}